R2-3134: Document CRC64NVME Support for Multipart Uploads (#23443)
cloudflare/cloudflare-docs: src/content/docs/r2/api/s3/api.mdx

This also applies to the `LocationConstraint` for the `CreateBucket` API.

## Checksum Types

> Checksums have an algorithm and a [type](https://docs.aws.amazon.com/AmazonS3/latest/userguide/checking-object-integrity.html#ChecksumTypes). All checksums don't necessarily support all types.

Reviewer: ```suggestion
Checksums have an algorithm and a [type](https://docs.aws.amazon.com/AmazonS3/latest/userguide/checking-object-integrity.html#ChecksumTypes). Note that not all checksum algorithms support both checksum types.
```
